spectator-api/src/main/java/com/netflix/spectator/impl/Scheduler.java

```diff
@@ -139,6 +144,12 @@ public Scheduler(Registry registry, String id, int poolSize) {
         .monitorValue(new AtomicInteger());
     taskExecutionTime = registry.timer(newId(registry, id, "taskExecutionTime"));
     taskExecutionDelay = registry.timer(newId(registry, id, "taskExecutionDelay"));
+    lastSuccessfulExecutionTimestamp = PolledMeter.using(registry)
+            .withId(newId(registry, id, "lastExecutionTimestamp", "result", "success"))
+            .monitorValue(new AtomicLong());
```

Comment:
This should be reporting the age not the timestamp. Also having both success/failure here doesn't really make sense to me. From the scheduler point of view it should be running regularly, the success or failure doesn't matter. Having both means normally when things are fine, if you look at the high level and do a max you'll typically get the failure since it will never get updated.

Maybe just a counter tracking uncaught exceptions that propagate. Many things will prevent the exception from throwing up so even though a task completes and doesn't throw it may still have failed from the application point of view.

Reply:
I'm assuming by age, you mean calculating the delta between monotonic time now and the last recorded monotonic time.

If we report the age instead of the timestamp, that won't catch the case we observed in which the task wasn't running at all... nothing would be running to update the age.  In that case it would look like the age since last success is constant and everything's okay.  I should switch to monotonic time, now that I think about it.

Maybe there's a better way to go about this.  Let's discuss in person.

Regarding the failure/exception comment... you make some good points.  I'll change it.